metals/src/main/scala/scala/meta/internal/parsing/FoldingRangeExtractor.scala

```diff
@@ -31,10 +30,12 @@ final class FoldingRangeExtractor(
   }
 
   def extractFrom(tree: Tree, enclosing: Position): Unit = {
-    if (span(tree.pos) > spanThreshold) {
-      val newEnclosing = tree match {
-        case Foldable((pos, adjust))
-            if span(enclosing) - span(pos) > distanceToEnclosingThreshold =>
+    // All Defn statements except one-liners must fold
+    if (
+      (tree.is[Defn] && span(tree.pos) > 0) || span(tree.pos) > spanThreshold
```

Comment:
`All Defn statements except one-liners must fold` <- I would say 3 and more like for everything. This should be a clear requirement across the board. Sorry if I was unclear previously.